integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/ExistingApiTestCase.java

```diff
@@ -77,4 +79,32 @@ public void testExistingApiWithSandboxKey() throws Exception {
         Assert.assertEquals(response.getResponseCode(), HttpStatus.SC_SUCCESS,
                 "Status code mismatched. Endpoint:" + endpoint + " HttpResponse ");
     }
+
+    @Test
+    public void checkHeadersSentToBackend() throws Exception {
+        Map<String, String> headers = new HashMap<>();
+        headers.put(TestConstant.AUTHORIZATION_HEADER, "Bearer " + accessTokenProd);
+
+        String endpoint = Utils.getServiceURLHttps(API_CONTEXT + "/1.0.0/headers");
+        HttpResponse response = HttpsClientRequest.retryGetRequestUntilDeployed(endpoint, headers);
+        Assert.assertNotNull(response, "Error occurred while invoking the endpoint " + endpoint + " HttpResponse ");
+        Assert.assertEquals(response.getResponseCode(), HttpStatus.SC_SUCCESS,
+                "Status code mismatched. Endpoint:" + endpoint + " HttpResponse ");
+        JSONObject responseJSON = new JSONObject(response.getData());
+        Assert.assertEquals(responseJSON.length(), 10, "Unexpected number of headers received by the backend");
+
+        Assert.assertNotNull(responseJSON.get("X-trace-key"));
+        Assert.assertNotNull(responseJSON.get("Accept"));
+        Assert.assertNotNull(responseJSON.get("X-request-id"));
+        Assert.assertNotNull(responseJSON.get("X-jwt-assertion"));
+        Assert.assertNotNull(responseJSON.get("X-forwarded-proto"));
+        Assert.assertNotNull(responseJSON.get("Host"));
+        Assert.assertNotNull(responseJSON.get("Pragma"));
+        Assert.assertNotNull(responseJSON.get("X-envoy-original-path"));
+        Assert.assertNotNull(responseJSON.get("User-agent"));
+        Assert.assertNotNull(responseJSON.get("Cache-control"));
+
+        Assert.assertFalse(responseJSON.has("x-wso2-cluster-header"));
+        Assert.assertFalse(responseJSON.has("x-envoy-expected-rq-timeout-ms"));
+    }
```

Comment:
Can we also add a test case to verify the headers received by the downstream client?

Reply:
Updating the PR to verify the headers received by the client. Fixed with https://github.com/wso2/product-microgateway/pull/2851/commits/ec34bc7665592ff70b09b6e4388f8d00a07ff9a0